src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1289,51 +1288,42 @@ internal static bool IsGenericTaskType(this TypeSymbol type, CSharpCompilation c
             {
                 return true;
             }
-            NamedTypeSymbol builderType;
-            MethodSymbol createBuilderMethod;
-            return namedType.IsCustomTaskType(out builderType, out createBuilderMethod);
+            object builderArgument;
+            return namedType.IsCustomTaskType(out builderArgument);
         }
 
         /// <summary>
-        /// Returns true if the type is generic or non-generic task-like type. If so, the async
-        /// method builder type is returned along with the method to construct that type.
+        /// Returns true if the type is generic or non-generic custom task-like type due to the
+        /// [AsyncBuilder(typeof(B))] attribute. It returns the "B".
         /// </summary>
-        internal static bool IsCustomTaskType(this NamedTypeSymbol type, out NamedTypeSymbol builderType, out MethodSymbol createBuilderMethod)
+        /// <remarks>
+        /// For the Task types themselves, this method might return true or false depending on mscorlib.
+        /// The definition of "custom task-like type" is one that has an [AsyncBuilder(typeof(B))] attribute,
+        /// no more, no less. Validation of builder type B is left for elsewhere. This method returns B
+        /// without validation of any kind.
+        /// </remarks>
+        internal static bool IsCustomTaskType(this NamedTypeSymbol type, out object builderArgument)
```

Comment:
@tmat It is an extension method because it doesn't feel like a core part of the `NamedTypeSymbol` API and it can be written against using the public API of `NamedTypeSymbol`.